llama_index/node_parser/extractors/metadata_extractors.py

```diff
@@ -332,3 +332,88 @@ def extract(self, nodes: Sequence[BaseNode]) -> List[Dict]:
                 metadata["section_summary"] = node_summaries[i]
 
         return metadata_list
+
+
+DEFAULT_ENTITY_MAP = {
+    "PER": "persons",
+    "ORG": "organizations",
+    "LOC": "locations",
+    "ANIM": "animals",
+    "BIO": "biological",
+    "CEL": "celestial",
+    "DIS": "diseases",
+    "EVE": "events",
+    "FOOD": "foods",
+    "INST": "instruments",
+    "MEDIA": "media",
+    "PLANT": "plants",
+    "MYTH": "mythological",
+    "TIME": "times",
+    "VEHI": "vehicles",
+}
+
+
+class EntityExtractor(MetadataFeatureExtractor):
+    """
+    Entity extractor. Extracts `entities` into a metadata field using a default model
+    `tomaarsen/span-marker-xlm-roberta-base-multinerd` and the SpanMarker library.
+
+    Install SpanMarker with `pip install span-marker`.
+
+    Args:
+        model_name: (Optional[Str]): The name of the SpanMarker model.
+        Default is `tomaarsen/span-marker-xlm-roberta-base-multinerd`.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "tomaarsen/span-marker-xlm-roberta-base-multinerd",
+        prediction_threshold: float = 0.5,
+        span_joiner: str = " ",
+        label_entities: bool = False,
+        device: Optional[str] = None,
+        entity_map: Optional[Dict[str, str]] = None,
+        tokenizer: Optional[Callable[[str], List[str]]] = None,
+    ):
+        try:
+            from span_marker import SpanMarkerModel
+        except ImportError:
+            raise ImportError(
+                "SpanMarker is not installed. Install with `pip install span-marker`."
+            )
+
+        try:
+            from nltk.tokenize import word_tokenize
+        except ImportError:
+            raise ImportError("NLTK is not installed. Install with `pip install nltk`.")
+
+        self._model = SpanMarkerModel.from_pretrained(model_name)
+        if device is not None:
+            self._model = self._model.to(device)
+
+        self._tokenizer = tokenizer or word_tokenize
+        self._prediction_threshold = prediction_threshold
+        self._span_joiner = span_joiner
+        self._label_entities = label_entities
+        self._entity_map = DEFAULT_ENTITY_MAP
```

Comment:
nit: entity_map arg isn't used

Reply:
also might be good to briefly add to docstring

Reply:
will do!

Reply:
(entity map is actually used in the line below, as well as in the extract function 👍🏻)

Reply:
docstring added!